bootstrap.py

```diff
@@ -61,190 +69,109 @@
         sys.exit("ERROR: Can't enable debug mode - Spyder is already imported")
-    print("*. Switching debug mode on")
+    logger.info("Switching debug mode on")
     os.environ["SPYDER_DEBUG"] = "3"
     if len(args.filter_log) > 0:
-        print("*. Displaying log messages only from the "
-              "following modules: {0}".format(args.filter_log))
+        logger.info("Displaying log messages only from the "
+                    "following modules: %s", args.filter_log)
     os.environ["SPYDER_FILTER_LOG"] = args.filter_log
     # this way of interaction suxx, because there is no feedback
     # if operation is successful
 
-
-# Add this path to the front of sys.path
-sys.path.insert(0, DEVPATH)
-print("*. Patched sys.path with %s" % DEVPATH)
-
-# Add external dependencies subrepo paths to be the next entries
-# (1, 2, etc) of sys.path
-DEPS_PATH = osp.join(DEVPATH, 'external-deps')
-i = 1
-for path in os.listdir(DEPS_PATH):
-    external_dep_path = osp.join(DEPS_PATH, path)
-    sys.path.insert(i, external_dep_path)
-    print("*. Patched sys.path with %s" % external_dep_path)
-    i += 1
-
-# Remove any leftover PyLSP installation from previous execution
-print("*. Removing previous PyLSP local installation.")
-pylsp_utils.remove_installation()
-
-# Install PyLSP locally
-print("*. Installing PyLSP locally")
-pylsp_utils.install()
-
-# Remove PyLSP temp installation at exit
-atexit.register(pylsp_utils.remove_installation)
-
-# Register completion providers
-print("*. Declaring completion providers")
-
-fallback = pkg_resources.EntryPoint.parse(
-    'fallback = spyder.plugins.completion.providers.fallback.provider:'
-    'FallbackProvider')
-
-snippets = pkg_resources.EntryPoint.parse(
-    'snippets = spyder.plugins.completion.providers.snippets.provider:'
-    'SnippetsProvider'
-)
-
-lsp = pkg_resources.EntryPoint.parse(
-    'lsp = spyder.plugins.completion.providers.languageserver.provider:'
-    'LanguageServerProvider'
-)
-
-# Create a fake Spyder distribution
-d = pkg_resources.Distribution(__file__)
-
-# Add the providers to the fake EntryPoint
-d._ep_map = {
-    'spyder.completions': {
-        'fallback': fallback,
-        'snippets': snippets,
-        'lsp': lsp,
-    }
-}
-
-# Add the fake distribution to the global working_set
-pkg_resources.working_set.add(d, 'spyder')
-
-
 # Selecting the GUI toolkit: PyQt5 if installed
 if args.gui is None:
     try:
         import PyQt5  # analysis:ignore
-        print("*. PyQt5 is detected, selecting")
+        logger.info("PyQt5 is detected, selecting")
         os.environ['QT_API'] = 'pyqt5'
     except ImportError:
         sys.exit("ERROR: No PyQt5 detected!")
 else:
-    print("*. Skipping GUI toolkit detection")
+    logger.info("Skipping GUI toolkit detection")
     os.environ['QT_API'] = args.gui
 
+# ---- Install sub repos
+
+if not args.no_subrepos:
+    for name in REPOS.keys():
+        if name == 'spyder':
+            continue
```

Comment:
Don't we want to ensure Spyder itself is installed too, as our script does?

Reply:
Sure. I'll remove that.